### PR TITLE
Add support for network/NatGateways

### LINKED ISSUE #2893: Feature: Support NAT Gateways
Example ARM template here: https://learn.microsoft.com/en-us/azure/virtual-network/nat-gateway/quickstart-create-nat-gateway-template

The actual resource is: [Microsoft.Network/natGateways](https://learn.microsoft.com/en-us/azure/templates/microsoft.network/natgateways)

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2893 

**What this PR does / why we need it**:

This PR adds support for `network/NatGateways` and `network/PublicIpPrefix`

Note: Did try adding `network/CustomIpPrefix` as well, however, looks like their API has some issues, and its getting stuck in `Provisioning` state too much in the background making it unable to delete. [Resource](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/82acd5bb-4206-47d4-9c12-a65db028483d/resourceGroups/asotest-rg-faseaf/providers/Microsoft.Network/customIpPrefixes/aso-sample-customprefix/overview) in portal.  [link](https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/manage-custom-ip-address-prefix#im-unable-to-delete-a-custom-ip-prefix) to docs stating about it getting stuck.

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/api/network/v1beta20220701/structure.txt
Comment (theunrepentantgeek) [inline v2/api/network/v1beta20220701/structure.txt:1]: Why is this file deleted?

**Review by theunrepentantgeek — APPROVED** [review]
Looks good - just one concern about a deleted file.

File: v2/api/network/v1beta20220701/structure.txt
Comment (super-harsh) [reply v2/api/network/v1beta20220701/structure.txt:1]: This was an empty folder with just structure.txt file in it, have a look at the link below. In my next commit(trying to get through the samples test) have also deleted the `v1beta20220701storage`. 

https://github.com/Azure/azure-service-operator/tree/main/v2/api/network/v1beta20220701
